### ISSUE TITLE
Tracking Releases of tldr-pages Corpus

### ISSUE THREAD

**Issue by SethFalco** [Helsinki-NLP/OPUS#28]
Good morning,

Back in 2023, I had written a tool and PR to include a tldr-pages as a corpus:

* https://github.com/tldr-pages/tldr-translation-pairs-gen
* https://github.com/Helsinki-NLP/OPUS-ingest/pull/22

I've noticed however that since it's addition, it has never been rerun to host OPUS. The tldr-pages project is constantly being updated, and we generate a new build of the translation pairs every month. (It's always hosted on the same URL, so the current `Makefile` should work.)

Size isn't everything, but just to illustrate the difference:

| Date | Corpus Size |
|--:|--:|
| September 2024 | 7.29 MB |
| October 2025 | 11.8 MB |

Is there anything that can be done to stay up-to-date with the latest release of the corpus?

**Comment by jorgtied** [comment]
Sorry for the delay: I just ran an update. Does this look correct? https://opus.nlpl.eu/legacy/tldr-pages.php
Unfortunately, I do not have a fully automatic procedure in place. I need to work on that but there is at least one step that requires manual input (upload to our storage requires some authentication that I need to check how I can automate it)

**Comment by SethFalco** [comment]
> Does this look correct?

LGTM! Thank you! 🎉

> Unfortunately, I do not have a fully automatic procedure in place

Ahh, understood!

It sounds like most of the road blocks are infra related, but I'll say it anyway. There's no rush of course, but when or if you do have the automation, don't be afraid to ping me (or any other maintainer) if there's anything we can do to make the process smoother for our dataset at least.

**Comment by jorgtied** [comment]
Thanks, I will keep that in mind. How often would it make sense to have a new release? I try to have make official releases so that you can refer to the data set version used in experiments. Would you consider that this data set is better be used as a test set than part of training data? I consider separating train and dev/test sets

**Comment by SethFalco** [comment]
> How often would it make sense to have a new release?

We do an automatic release once a month (arbitrary), so if you wanted to peg to that, then monthly would make sense if it could be automated and did not impose significant cost/hassle for you.

In general, though, I think doing at least once release a year is sensible, though. Ofc, anything in between is welcome!

> Would you consider that this data set is better be used as a test set than part of training data? I consider separating train and dev/test sets

Hmm, I didn't realize this is something you considered doing on your end. I assumed which datasets would be training vs. testing would have been down to a downstream decision, especially in cases someone wanted different ratios per source. 🤔

Setting aside the small size of our dataset (relative to Wikipedia, GNOME, etc.) I think ours is good for training as it's very densely packed with technical language. (Programming and UNIX terms, etc.)

_Feel free to shout at me if you think I've misunderstood you!_
